src/clojure_lsp/main.clj

```diff
@@ -430,6 +431,24 @@
         (end
           (apply #'handlers/extension method args)))))
 
+(defn process-alive?
+  [pid]
+  (let [{:keys [exit]} (shell/sh "kill" "-0" (str pid))]
+    (= exit 0)))
+
+(defn start-parent-process-liveness-probe!
+  [ppid server]
+  (future (run!
+           (fn [_]
+             (Thread/sleep 5000)
+             (log/info "Checking parent process" ppid "liveness")
```

Comment:
I'd change this one to avoid a lot of logs (I have an idea to remove debug logs on the production relese)
```suggestion
             (log/debug "Checking parent process" ppid "liveness")
```